test/functional/spec-runner/index.js

```diff
@@ -307,22 +314,24 @@ function runTestSuiteTest(configuration, spec, context) {
 
     let session0, session1;
     let savedSessionData;
-    try {
-      session0 = client.startSession(
-        Object.assign({}, sessionOptions, parseSessionOptions(spec.sessionOptions.session0))
-      );
-      session1 = client.startSession(
-        Object.assign({}, sessionOptions, parseSessionOptions(spec.sessionOptions.session1))
-      );
 
-      savedSessionData = {
-        session0: JSON.parse(EJSON.stringify(session0.id)),
-        session1: JSON.parse(EJSON.stringify(session1.id))
-      };
-    } catch (err) {
-      // ignore
-    }
+    if (context.useSessions) {
```

Comment:
Did you try optionally creating these if the entities exist? (i.e. if `session0` or `session1` are specified) It seems like a preferable solution to feature flagging in this particular case. I was recommending the feature flag for cases which couldn't be resolved this way (like skip init)

Reply:
I did try to create them if the entities exist, it created a bug in the transactions suite. I had to dig deep into it but apparently those test require the object to exist in order to work correctly.

Reply:
I just tried again, the transaction tests do not have `spec.sessionOptions` set but do require a session, that code to run.